timeseries/src/autogluon/timeseries/models/ensemble/greedy.py

```diff
@@ -98,22 +130,28 @@ def _calculate_regret(self, y_true, y_pred_proba, metric=None, sample_weight=Non
         return -avg_score
 
 
-class TimeSeriesGreedyEnsemble(AbstractTimeSeriesEnsembleModel):
-    """Constructs a weighted ensemble using the greedy Ensemble Selection algorithm."""
+class TimeSeriesGreedyEnsemble(AbstractWeightedTimeSeriesEnsembleModel):
+    """Constructs a weighted ensemble using the greedy Ensemble Selection algorithm by
+    Caruana et al. [Car2004]
+
+    References
+    ----------
+    .. [Car2024] Caruana, Rich, et al. "Ensemble selection from libraries of models."
+        Proceedings of the twenty-first international conference on Machine learning. 2004.
+    """
 
     def __init__(self, name: Optional[str] = None, ensemble_size: int = 100, **kwargs):
         if name is None:
-            name = "WeightedEnsemble"
+            name = "GreedyWeightedEnsemble"
```

Comment:
Will this result in the model being called `GreedyWeightedEnsemble` in the `TimeSeriesTrainer` model graph? If yes, this could be a breaking change for some users who expect the `WeightedEnsemble` to be the name of the final ensemble model in AG. 

Let's add a TODO here and revisit this once we expose the API for ensemble configuration in v1.4+?

Reply:
OK reverting this.